backend/hf_service.py

```
@@ -34,28 +34,27 @@ async def query_hf_api(image_bytes, labels, client=None):
-            return response.json()
-        except httpx.HTTPError as e:
-            logger.error(f"HF API HTTP Error: {e}")
-            raise ExternalAPIException("Hugging Face API", str(e)) from e
-        except Exception as e:
-            logger.error(f"HF API Request Exception: {e}")
-            raise ExternalAPIException("Hugging Face API", str(e)) from e
+    try:
+        response = await client.post(API_URL, headers=headers, json=payload, timeout=20.0)
+        if response.status_code != 200:
+            logger.error(f"HF API Error: {response.status_code} - {response.text}")
+            raise ExternalAPIException("Hugging Face API", f"HTTP {response.status_code}: {response.text}")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)